### DISCUSSION TITLE
Welcome to Linear_Feedback_Shift_Register Discussions!

### DISCUSSION THREAD

**Discussion by Nikeshbajaj** [Nikeshbajaj/Linear_Feedback_Shift_Register#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Tayyabali535** [comment]
hi there, I am new in python and I do not know how to install or use anything, can you please guide me how to run your tool, thank you

**Comment by Nikeshbajaj** [comment]
Hi @Tayyabali535 

The full documentation is **[https://lfsr.readthedocs.io/](https://lfsr.readthedocs.io/)**

Here is a quick startup; assuming you know a little about python and numpy

1. First install pylfsr using  `pip install pylfsr`.  (Details can be found here **[on the docs](https://lfsr.readthedocs.io/en/latest/installation.html#installation)**
2. A simple example for generating sequence from a 5-bit LFSR is here

    Run following on a python terminal, IPython, or  on a jupyter-notebook

    ```
    import numpy as np
    from pylfsr import LFSR
    
    
    state = [0,0,0,1,0]
    fpoly = [5,4,3,2]
    L = LFSR(fpoly=fpoly,initstate =state, verbose=True)
    L.info()
    tempseq = L.runKCycle(10)
    ```

    Other basic examples of LFSR can be found here  - **[Basic Examples](https://lfsr.readthedocs.io/en/latest/LFSR_Examples.html#basic-examples)**


3. Further to display LFSR  run  `L.Viz(title='R1')`

   More details on discplaying and visualization, can be found here **[Visualizationn](https://lfsr.readthedocs.io/en/latest/dispViz.html#)**
